libdnf5/module/module_sack.cpp

```diff
@@ -555,6 +558,27 @@ std::pair<std::vector<std::vector<std::string>>, ModuleSack::ModuleErrorType> Mo
 }
 
 
+GoalProblem ModuleSack::Impl::module_solver_error_to_goal_problem(ModuleSack::ModuleErrorType module_error) {
```

Comment:
Subject for a discussion - The last changes makes somehow `ModuleErrorType` redundant because GoalProblem has the same item set. Or did I missed something. Removal of ModuleErrorType would result in change of IP, but there is 5.2 tah introduce significant changes.

Other option would be that GoalProblem might inherit items from `ModuleErrorType`. This option might be a problem in future when module will get optional.

And the last option we can keep it like it is.